services/app-web/src/components/SubmissionSummarySection/ContractDetailsSummarySection/ContractDetailsSummarySection.tsx

```diff
@@ -79,9 +82,44 @@ export const ContractDetailsSummarySection = ({
         itemsAmendedOtherList.push(amendedOtherReason)
     }
 
+    // Get the zip file for the contract
+    const { getKey, getBulkDlURL } = useS3()
+    useEffect(() => {
+        // get all the keys for the documents we want to zip
+        async function fetchZipUrl() {
+            const keysFromDocs = submission.contractDocuments
+                .map((doc) => {
+                    const key = getKey(doc.s3URL)
+                    if (!key) return ''
+                    return key
+                })
+                .filter((key) => key !== '')
+
+            // call the lambda to zip the files and get the url
+            const zippedURL = await getBulkDlURL(
+                keysFromDocs,
+                submission.name + '-contract-details.zip'
+            )
+            if (zippedURL instanceof Error) {
+                console.log('ERROR: TODO: DISPLAY AN ERROR MESSAGE')
+                return
+            }
+
+            setZippedFilesURL(zippedURL)
+        }
+
+        void fetchZipUrl()
+    }, [getKey, getBulkDlURL, submission])
+    const [zippedFilesURL, setZippedFilesURL] = useState<string>('')
```

Comment:
I think I prefer that all useState() calls be at the top of the function just to keep them near the props.